src/functionalTest/kotlin/com/github/jengelman/gradle/plugins/shadow/ShadowPluginTest.kt

```diff
@@ -765,8 +769,15 @@ class ShadowPluginTest : BasePluginTest() {
 
     run(shadowJarTask)
 
-    val entries = outputShadowJar.use { it.entries().toList() }
-    assertThat(entries.count { it.name.endsWith(".class") }).isEqualTo(1)
+    assertThat(outputShadowJar).useAll {
+      transform { actual -> actual.entries().toList().map { it.name }.filter { it.endsWith(".class") } }
+        .single().isEqualTo("my/plugin/MyPlugin.class")
+      transform { it.manifest.mainAttributes.keys }.all {
+        isNotEmpty()
+        // Doesn't contain Gradle classes.
+        doesNotContain("Class-Path")
```

Comment:
This may not work properly. The key set is `Set<Object>` instead of `Set<String>`.

Reply:
Fixing in #1161.